### PR TITLE
Wizard time

### PR DESCRIPTION
**Pull request by ronlawrence3**
@BarbourSmith  

 I'll publish an index.html.gz for people to try and if it is well received we can merge it or change it based on feedback. It creates a wizard for setting up the maslow in another tab. The tab is activated if the maslow is not homed, but you can navigate to / from it also.

image
image
image
image
image
image

also:
image

### CODE REVIEW

**Comment by BarbourSmith** [comment]
This looks fantastic! I love the idea. I’m moving today so I’m going to be away from my machine, but I will play around with it first thing tomorrow

File: www/js/tablet.js [hunk 1]
```diff
@@ -455,18 +423,23 @@ function tabletShowMessage(msg, collecting) {
...
   }
   if (msg.startsWith('error:')) {
-    const msgExtra = {
-      "8": " - Command requires idle state. Unlock machine?",
-      "152": " - Configuration is invalid. Maslow.yaml file may be corrupt. Try restarting",
-      "153": " - Configuration is invalid. ESP32 probably did a panic reset. Config changes cannot be saved. Try restarting",
-    };
-
-    msg += msgExtra[msg.split(":")[1]] || "";
+    let parts = msg.split(":");
```
Comment (BarbourSmith) [inline www/js/tablet.js]: It looks like we might have the issue where some of these changes are undoing changes on the main branch. @md8n cleaned up some of this code here

Comment (ronlawrence3) [reply www/js/tablet.js]: I'll have another look at my diffs and resolve this. I just noticed the merge I did had issues when I ran it so likely I just resolved things wrong.

Comment (BarbourSmith) [reply www/js/tablet.js]: That always gets tricky

**Comment by ronlawrence3** [comment]
More testing needed... I'll see if I can run this through a couple cycles in the next few days...

**Comment by ronlawrence3** [comment]
Also today I decided to give this a spin outside the python proxy by uploading, and it won't fit on the filesystem, so I will have more to do to slim it down some.

**Comment by ronlawrence3** [comment]
OK, I think this is in good shape now and worth a try. Feedback welcome!

**Comment by ronlawrence3** [comment]
oops, forgot to @BarbourSmith  notify I'm ready :) also no rush... just want to make sure you know its ready to look at

**Comment by BarbourSmith** [comment]
I haven't forgotten. I think that this one will take a decent amount of time to test and I want to track down this crashing bug first, but I won't forget!

**Comment by ronlawrence3** [comment]
I totally understand and fully agree with your priorities! I did a bit more polishing of the code today. it is still behaving pretty well.

**Comment by BarbourSmith** [comment]
I'm finally having a second to really dig into this and I think I have a plan. 

The goal of wizard is to guide the user through the calibration process, but I think that what we really need to do is make the process simpler to the point that we don't need a wizard. 

I have a plan that I'd like to work towards which would be to have just two buttons one for "Extend Belts" and one for "Retract Belts".

The extend belts just extends the belts as long as you are pulling on them. We don't ask the user the size of their frame, we just let them extend enough to attach to the frame.

The retract belts pulls the belts tight. When the belts get tight we can check to see if everything looks good with calibration. If the belt lengths match what we would expect from the stored calibration then we're good, if they don't match up then we can ask the user if they want to run calibration. 

This would also let us store the belt lengths and re-use them without needing to do the whole Retract -> Extend cycle and not worry about breaking the belts because the first thing that the retract button does is pull the belts tight and check that everything is OK which would catch cases where the belts had moved while the machine was off.

Does that seem like a reasonable alternative?

**Comment by ronlawrence3** [comment]
> Does that seem like a reasonable alternative?

Absolutely! I'm lukewarm on my implementation of the wizard anyway after living with it a while, and the forums have calmed down a lot in terms of calibration confusion, so this is probably not needed any more. 

I have other thoughts too, such as once the machine is on the frame, detecting horizontal or vertical, detecting what threshold to set, etc.

**Comment by BarbourSmith** [comment]
> I have other thoughts too, such as once the machine is on the frame, detecting horizontal or vertical, detecting what threshold to set, etc.

These sound like fantastic ideas. Detecting horizontal or vertical is top of my list for things that I think we need.

What do you mean by detecting the threshold? How would that work?

**Comment by ronlawrence3** [comment]
> What do you mean by detecting the threshold? How would that work?

This is just my naive thinking probably, but when first assembled, the retract (for me at least) had to be increased until it would retract fully.  I've seen others report this as well. I assume it is however tight the arms are, etc. 

I was thinking an initial step after assembly could be retract-all at something like 2500, then extend out maybe 100mm on each arm and have the maslow start low and retract each for a time and then stop and check if it did retract, then repeat until meeting the threshold. Not sure this would work, but that is what I was thinking.

**Comment by BarbourSmith** [comment]
I love that idea. I think that would work perfectly.

**Pull request closed without merging**